roles/marmotta/tasks/tomcat-marmotta.yml

```diff
@@ -22,34 +23,42 @@
 
 - name: install marmotta
   become: yes
-  command: cp {{ install_path }}/apache-marmotta-{{ marmotta_ver }}/marmotta.war /var/lib/tomcat/webapps/marmotta.war \
-           creates=/var/lib/tomcat/webapps/marmotta.war
+  command: cp {{ install_path }}/apache-marmotta-{{ marmotta_ver }}/marmotta.war {{ tomcat }}/webapps/marmotta.war \
+           creates={{ tomcat }}/webapps/marmotta.war
 
 - name: set ownership on marmotta.war
   become: yes
-  command: chown tomcat:tomcat /var/lib/tomcat/webapps/marmotta.war
+  file: path={{ tomcat }}/webapps/marmotta.war owner=tomcat group=tomcat
 
 - name: set permissions for marmotta_home
   become: yes
   file: path={{ marmotta_home }} owner=marmotta group=marmotta mode=0755 state=directory recurse=yes
 
-- name: set marmotta system config properties (linux)
+- name: set marmotta system config properties
   become: yes
   template: src=system-config.properties.j2 dest={{ marmotta_home }}/system-config.properties \
             owner=tomcat group=tomcat backup=no mode=0600
 
+- name: restart tomcat
+  service: name=tomcat enabled=yes state=restarted
+
+- wait_for: path={{ tomcat }}/webapps/marmotta/WEB-INF/lib/log4j-over-slf4j-1.7.7.jar
+
 # Currently Marmotta tries to install a
 # logging library that conflicts with Solr's, so for now we're just
-# removing that library. 
+# removing that library.
 # Check - is the library installed before Marmotta runs?
 # If not, restart tomcat after installing Marmotta, then remove
 # the library, then restart Tomcat again.
 
 - name: remove log4j-over-slf4j.jar
-  file: path={{ item }} state=absent
   become: yes
+  file: path={{ item }} state=absent
   with_items:
-    - /var/lib/tomcat/webapps/marmotta/WEB-INF/lib/log4j-over-slf4j-1.7.7.jar
+    - "{{ tomcat }}/webapps/marmotta/WEB-INF/lib/log4j-over-slf4j-1.7.7.jar"
+
+# Otherwise subsequent runs will hang on the "wait_for" above
```

Comment:
The ubuntu roles have separate segments for Install and Configure, which allows updating existing machines more efficiently because you're not including all the installation tests. I'm not suggesting changes to this PR, just a comment on a relevant solution to issues with subsequent Ansible runs.